magneto_lib/magneto.h

```diff
@@ -1,10 +1,37 @@
 #ifndef MAGNETO_LIB_MAGNETO_H
 #define MAGNETO_LIB_MAGNETO_H
-
+#include <map>
 namespace cr {
 namespace magneto {
 
-// Defines library API
+ enum  TypeOfPiece {
+        king,          
+        queen,
+        rook,
+        pawn,       
+        bishop,    
+        knight,
+         
+    };
+
+    enum Color{
+        white,
+        black,
+    };
+
+    struct Square{
+        int row;
+        int column;
+        std::map <int, std::map<int,std::string>> map;
```

Comment:
Several notes here.

- Square is supposed to be a lightweight object, and having a two-level map is quite the opposite of that.
- Why store all 64 strings for all combinations if we can simply build it and return when asked?
- We probably want the Square to be immutable, to encapsulate the row and the column and provide a read-only access (getters).
- Maybe rename them to "file" and "rank"?
- We also want to have no default ctor for Square and only one with two params, file and rank. We don't want the user to create a random square or a default square. Let them specify which rank and file.

```
Square sq; // Error
Square sq(3, 5); // OK, creates an index for "d4" square
Square sq("d4"); // OK, the same as above
Square sq(293, 532); // Warning, creates a square with {8, 8}, logs an error or warning. (This is because we decided not to go with exceptions)
```